### PR TITLE
Briefly document flow-insensitive analysis

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

File: src/analyses/README.md [hunk 1]
```diff
@@ -18,7 +18,16 @@ This is obsolete.
...
-To be documented.
+Framework for flow-insensitive analyses. Maintains a single global domain which
+instructions are invited to transform in turn. Unsound (terminates too early) if
+(a) there are two or more instructions that incrementally reach a fixed point,
+for example by walking a chain of pointers and updating a points-to set, but
+(b) those instructions are separated by instructions that don't update the
+domain (for example, SKIP instructions) unless the domain carefully avoids this
+fate. Therefore, not recommended for new code.
+
+Only current user in-tree is \ref value_set_analysis_fit and its close
```
Comment (chrisr-diffblue) [inline src/analyses/README.md]: Is `flow_insensitive_analysist` still active, or has it been/will it be deprecated like `static_analysist` ? If so should we clearly state that here? (there is the ambiguous "Therefore, not recommended for new code.", but perhaps it should be more forceful)

Comment (smowton) [reply src/analyses/README.md]: AFAICT it is not explicitly deprecated, that's just my recommendation

**Review by chrisr-diffblue — APPROVED** [review]
Thanks for clarifying. LGTM

File: src/analyses/README.md [hunk 2]
```diff
@@ -18,7 +18,16 @@ This is obsolete.
 
 \subsection analyses-flow-insensitive-analysis Flow-insensitive analysis (flow_insensitive_analysist)
 
-To be documented.
+Framework for flow-insensitive analyses. Maintains a single global domain which
```
Comment (peterschrammel) [inline src/analyses/README.md]: Use 'abstract value' instead of 'domain' (several occurrences).

**Comment by smowton** [comment]
@peterschrammel done in the readme, in the header file I defined what I meant by "the domain"

**Pull request merged**